File: schemas/preorder.py
from pydantic import BaseModel
from typing import Optional

class PreorderModel(BaseModel):
    seller: str
    models: str
    eta: Optional[str] = None
    total_price: Optional[float] = None
    po_amount: Optional[float] = None
    on_arrival_amount: Optional[float] = None
    delivery_status: Optional[str] = "Pending"

class PreorderRead(PreorderModel):
    serial_number: int
    date_added: Optional[str] = None

File: models/Preorder.js
const mongoose = require('mongoose');

const preorderSchema = new mongoose.Schema({
  serial_number: { type: Number, required: true, unique: true, index: true },
  seller: { type: String, index: true },
  seller_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Seller', index: true },
  seller_link: { type: String },
  models: { type: String, required: true },
  eta: { type: String },
  total_price: { type: Number },
  po_amount: { type: Number },
  on_arrival_amount: { type: Number },
  delivery_status: { type: String, default: "Pending" },
  date_added: { type: String }, // Stored as YYYY-MM-DD for compatibility
  user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: false, index: true }
}, { timestamps: true });

module.exports = mongoose.model('Preorder', preorderSchema);
